pkg/master/controller/crdregistration/conversion.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crdregistration
+
+import (
+	"encoding/json"
+
+	"github.com/go-openapi/spec"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+// Convert_apiextensions_JSONSchemaProps_To_spec_Schema converts our internal OpenAPI v3 schema
+// (*apiextensions.JSONSchemaProps) to an OpenAPI v2 schema (*spec.Schema).
+func Convert_apiextensions_JSONSchemaProps_To_spec_Schema(in *apiextensions.JSONSchemaProps) (*spec.Schema, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	// Convert internal JSONSchemaProps to versioned JSONSchemaProps, which has the JSON tags
+	versioned := new(v1beta1.JSONSchemaProps)
+	if err := v1beta1.Convert_apiextensions_JSONSchemaProps_To_v1beta1_JSONSchemaProps(in, versioned, nil); err != nil {
+		return nil, err
+	}
+
+	// Marshal JSONSchemaProps into JSON and unmarshal the data into spec.Schema
+	data, err := json.Marshal(*versioned)
+	if err != nil {
+		return nil, err
+	}
+	out := new(spec.Schema)
+	if err := out.UnmarshalJSON(data); err != nil {
+		return nil, err
+	}
+	// Remove unsupported fields in OpenAPI v2
+	out.AllOf = nil
```

Comment:
@nikhita had a nice spreadsheet with the differences.

Reply:
@nikhita Could you point me to it? Thanks!